packages/adminrouter/extra/src/lib/cache.lua

```diff
@@ -166,9 +166,12 @@ local function fetch_and_store_marathon_apps(auth_token)
           end
        end
 
-       -- next() returns nil if table is empty.
-       local i, task = next(tasks)
-       if i == nil then
+       -- Randomly select the task.
+       math.randomseed(os.time())
```

Comment:
A pseudorandom number generator (PRNG) is supposed to be seeded _once_ for many subsequent pseudorandom number generations.

In fact, that's the designed way for a PNRG to yield a uniform distribution of pseudorandom numbers.

Seeting it before every number generation probably yields a non-uniform distribution.

Reply:
Good. Do you think the function refresh_mesos_state_cache (in cache empty and cache expired) would be a good place to initialize it?

Reply:
What do you think about the beginning of `periodically_refresh_cache` function[1]? It runs in `init_worker_by_lua` context [2].

[1] https://github.com/dcos/dcos/blob/5c019cfdee6a17ea4b2252e7c0e98118bc845a06/packages/adminrouter/extra/src/lib/cache.lua#L552
[2] https://github.com/openresty/lua-nginx-module#init_worker_by_lua

Reply:
Done. Thanks!